src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -499,32 +510,43 @@ def save_as_table(
         child: SnowflakePlan,
     ) -> SnowflakePlan:
         if mode == SaveMode.APPEND:
-            create_table = create_table_statement(
-                table_name,
-                attribute_to_schema_string(child.attributes),
-                error=False,
-                table_type=table_type,
-            )
-
-            return SnowflakePlan(
-                [
-                    *child.queries[0:-1],
-                    Query(create_table),
-                    Query(
-                        insert_into_statement(
-                            table_name=table_name,
-                            child=child.queries[-1].sql,
-                            column_names=column_names,
-                        )
+            if self._table_exists(table_name):
+                return self.build(
```

Comment:
in this case, why is `child.queries` are not considered anymore?

Reply:
Nice catch. It should also have `child, None` as other modes.

Reply:
Nice catch. It should also have `child, None` as other modes.

Reply:
Wait it already has `child, None`, just not very obvious in the diff. I assume `child` should have `child.queries` handled?

Reply:
oops bad github!

Reply:
Yeah totally...I also did not see that in github, and stared at my pycharm questioning the meaning of life :/